Python/wrfrun/unCMIP6.py

```diff
@@ -0,0 +1,622 @@
+                    lvlmark = ''
+                # Make initial part of file name
+                fn_pre = input_root+'/'+varname+'_'+frq+lvlmark+'_'+self.model_name
+                fn_pre = fn_pre+'_'+self.exp_id+'_'+self.esm_flag+'_'+self.grid_flag  
+                # Search for files
+                filenames = sorted(glob.glob(fn_pre+'*.nc'))
+                dates = [ele[len(fn_pre):-3] for ele in filenames]
+                startdates_t = []
+                enddates_t = []
+                for ele in dates:
+                    ele_split = ele.split("-")
+                    startdates_t.append(ele_split[0])
+                    enddates_t.append(ele_split[1])
+                startdates = []
+                enddates = []
+                if len(startdates_t[0])==12:
+                    dformat = '%Y%m%d%H%M'
+                    dlen = 12
+                elif len(startdates_t[0])==8:
+                    dformat = '%Y%m%d'
+                    dlen = 8
+                elif len(startdates_t[0])==6:
+                    dformat = '%Y%m'
+                    dlen = 6
+                else:
+                    raise ValueError('Error: File date format not recognized.')
+                for ele in startdates_t:
+                    if len(ele)!=dlen:
+                        raise ValueError('Error: Files do not have a consistent date format.')
+                    startdates.append(datetime.datetime.strptime(ele,dformat))    
+                for ele in enddates_t:
+                    if len(ele)!=dlen:   
+                        raise ValueError('Error: Files do not have a consistent date format.')
+                    enddates.append(datetime.datetime.strptime(ele,dformat))        
+                ndatefound = 0
+                for i in range(len(startdates)):
+                    if ((startdates[i]<=self.input_date) and (self.input_date<=enddates[i])):
+                        ndatefound = ndatefound+1
+                        filestartdate = startdates[i]
+                        fileenddate = enddates[i]
+                if ndatefound!=1:
+                    if not(itm['approx_dates']):
+                        raise ValueError('Error: Item is needed at exact dates, but number of files found is not 1.')
+                    else:
+                        if ndatefound>1:
+                            raise ValueError('Error: Number of files containing item is greater than 1.')
+                        else:
+                            if self.input_date<startdates[0]:
+                                filestartdate = startdates[0]
+                                fileenddate = enddates[0]
+                            elif enddates[-1]<self.input_date:
+                                filestartdate = startdates[-1]
+                                fileenddate = enddates[-1]
+                            else:
+                                ndatefound2 = 0
+                                for i in range(len(startdates)-1):
```

Comment:
I'm not sure if this case would normally occur... only if the data record was discontinuous, which would normally be unintentional and maybe it would be better to raise an error? Or are there reasons why we would have a discontinuous data record for some variables?

Reply:
We talked about this. File dates are "discontinuous".